plugins/module_utils/netbox_extras.py

```diff
@@ -34,9 +39,80 @@ def run(self):
         data = self.data
 
         # Used for msg output
-        name = data.get("name")
+        if data.get("name"):
+            name = data["name"]
+
+        if data.get("regions"):
+            regions = data["regions"]
+            nb_app = getattr(self.nb, "dcim")
+            nb_region = getattr(nb_app, "regions")
+            regionids = []
+            for region in regions:
+                regionids.append(nb_region.get(slug=(region)).id)
+                data["regions"] = regionids
+
+        if data.get("sites"):
+            sites = data["sites"]
+            nb_app = getattr(self.nb, "dcim")
+            nb_site = getattr(nb_app, "sites")
+            siteids = []
+            for site in sites:
+                siteids.append(nb_site.get(slug=(site)).id)
+                data["sites"] = siteids
+
+        if data.get("roles"):
+            roles = data["roles"]
+            nb_app = getattr(self.nb, "dcim")
+            nb_device_role = getattr(nb_app, "device_roles")
+            roleids = []
+            for role in roles:
+                roleids.append(nb_device_role.get(slug=(device_role)).id)
+                data["roles"] = roleids
+
+        if data.get("platforms"):
+            platforms = data["platforms"]
+            nb_app = getattr(self.nb, "dcim")
+            nb_platform = getattr(nb_app, "platforms")
+            platformids = []
+            for platform in platforms:
+                platformids.append(nb_platform.get(slug=(platform)).id)
+                data["platforms"] = platformids
+
+        if data.get("cluster_groups"):
+            cluster_groups = data["cluster_groups"]
+            nb_app = getattr(self.nb, "virtualization")
+            nb_cluster_group = getattr(nb_app, "cluster_groups")
+            cluster_groupids = []
+            for cluster_group in cluster_groups:
+                cluster_groupids.append(nb_cluster_group.get(slug=(cluster_group)).id)
+                data["cluster_groups"] = cluster_groupids
+
+        if data.get("clusters"):
+            clusters = data["clusters"]
+            nb_app = getattr(self.nb, "virtualization")
+            nb_cluster = getattr(nb_app, "clusters")
+            clusterids = []
+            for cluster in clusters:
+                clusterids.append(nb_cluster.get(name=(cluster)).id)
+                data["regions"] = regionids
+
+        if data.get("tenant_groups"):
+            tenant_groups = data["tenant_groups"]
+            nb_app = getattr(self.nb, "tenancy")
+            nb_tenant_group = getattr(nb_app, "tenant_groups")
+            tenant_groupids = []
+            for tenant_group in tenant_groups:
+                tenant_groupids.append(nb_tenant_group.get(slug=(tenant_group)).id)
+                data["tenant_groups"] = tenant_groupids
 
-        data["slug"] = self._to_slug(name)
+        if data.get("tenants"):
```

Comment:
I think we should most these to resolving via the normal methods. This would require adding the plurals to the `CONVERT_TO_ID` and then `ALLOWED_QUERY_PARAMS` for each.

This should allow people to specify them either via string or dictionary to specify extra context to find the specific endpoint.

Reply:
@FragmentedPacket Mikhail, integration test were added, all assertions passed for netbox 2.8/2.9. As of resolving, could you merge it for now as it is, and I'll try to modify the code as suggested?

Reply:
I'd rather not merge something that still needs work. I can take this over and get it across the finish line since I have a lack of contribution docs.

I appreciate all the heavy lifting you have done for the two modules! I'm going to do the same for the `netbox_tag` PR as it just needs minor work that I can get done tonight.

Again I really appreciate your help on these and I'll leave you commits attached to make sure you get credit for these!